### PR TITLE
Javascript/javascript2/week3

### PR DESCRIPTION
**Pull request by cph-kiwi**
I have completed my javascript/javascript2/week3 homework.

### CODE REVIEW

File: javascript/javascript2/week3/warmup.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+runAfterDelay(4, function () {
+  console.log("should be logged after 4 seconds");
+});
+
+// Check if we have double clicked on the page. A double click is defined by two clicks within 0.5 seconds. If a double click has been detected, log out the text: "double click!"
+let lastClick;
+
+document.querySelector("body").addEventListener("click", function () {
```
Comment (benna100) [inline javascript/javascript2/week3/warmup.js:91]: Super nice!

**Review by benna100 — CHANGES_REQUESTED** [review]
Okay I'm glad we got you into this class. Your work is **very** impressive! Good variable names, good at creating function, nice structure, well thought out.

Really well done! So think about function responsibility. What should a function have as parameters (what makes sense so it is easy and versatile to use) and what should the function do. A function should at best case only do one thing (Single responsibility principle). And keep an eye out for sideeffects. They sneak in everywhere!

File: javascript/javascript2/week3/fastest-presser.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
+let interval;
+let playerOne = 0;
+let playerTwo = 0;
+const playerOneScoreElement = document.getElementById("count-s");
+const playerTwoScoreElement = document.getElementById("count-l");
+const timerInput = document.getElementById("timer");
+
+const buttonElement = document.getElementById("button");
```
[the remark this answers is not in the document]
Comment (cph-kiwi) [reply javascript/javascript2/week3/fastest-presser.js]: Ok. Can do.

File: javascript/javascript2/week3/fastest-presser.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+buttonElement.textContent = "Start game!";
+const message = document.getElementById("message");
+message.textContent = "Set seconds and start game by clicking on the button.";
```
[the remark this answers is not in the document]
Comment (cph-kiwi) [reply javascript/javascript2/week3/fastest-presser.js:11]: Good point.

File: javascript/javascript2/week3/fastest-presser.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+function createConfetti(id) {
```
[the remark this answers is not in the document]
Comment (cph-kiwi) [reply javascript/javascript2/week3/fastest-presser.js:13]: Yeah, this one was tricky for me to think through. Once I had figured out how to achieve the desired result, I didn't want to disturb it by trying to fix it up. I'll have a go.

File: javascript/javascript2/week3/fastest-presser.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+function removeConfetti() {
+  const canvas = document.getElementById("my-canvas");
+  if (canvas === null) {
+    return;
+  }
+  canvas.parentNode.removeChild(canvas);
+}
+
+function updatePlayerScore(e) {
```
[the remark this answers is not in the document]
Comment (cph-kiwi) [reply javascript/javascript2/week3/fastest-presser.js:34]: Ok, I'll incorporate that.

File: javascript/javascript2/week3/function-as-a-variable.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+console.log(firstNumber());
+console.log(secondNumber());
+
+// Create an object that has a key whose value is a function. Try calling this function.
+const objectOne = {
+  id: 1,
+  name: "First Example Object",
+  functionKey: function () {
```
[the remark this answers is not in the document]
Comment (cph-kiwi) [reply javascript/javascript2/week3/function-as-a-variable.js:40]: Ok.

**Comment by cph-kiwi** [comment]
I have incorporated reviewers suggestions into the homework.

**Pull request merged**